### ISSUE TITLE
what is the least-necessary access for a Promscale role?

### ISSUE THREAD

**Issue by ssube** [timescale/promscale#580]
I'm trying to run Promscale against a Timescale Forge instance using a limited, non-superuser for regular operation. What is the least-necessary set of permissions required to run the Promscale for reading and writing to metrics in the `tsdb` database?

Some details:

- these instances will always run with `-install-extensions=false -migrate=false -upgrade-extensions=false`
  - migrations will be run by an admin through CI jobs (at no point will humans have access to passwords)
- we will probably be creating the role with TF, but would like to use Vault dynamic users if possible (and manage all roles/grants through one of those)
- we will be running on Timescale Forge instances

I am receiving the following error when trying to connect, after creating a role and granting it usage/select on the public schema:

```
> kubectl -n monitoring logs -f promscale-gitlab-staging-585cc77785-77kbt
level=info ts=2021-04-09T17:08:09.043Z caller=runner.go:32 msg="Version:0.3.0; Commit Hash: "
level=info ts=2021-04-09T17:08:09.043Z caller=runner.go:33 config="&{ListenAddr::9201 PgmodelCfg:{CacheConfig:
  {SeriesCacheInitialSize:250000 seriesCacheMemoryMaxFlag:{kind:0 value:50} SeriesCacheMemoryMaxBytes:429496729 
  MetricsCacheSize:10000 LabelsCacheSize:10000} AppName:promscale@0.3.0 Host:xxx.yyy.tsdb.forge.timescale.com Port:34139 
  User:cluster_gitlab_staging password:**** Database:tsdb SslMode:require DbConnectRetries:0 DbConnectionTimeout:1m0s 
  AsyncAcks:false ReportInterval:0 WriteConnectionsPerProc:4 MaxConnections:-1 UsesHA:false DbUri: 
  EnableStatementsCache:true HAEnabled:false} LogCfg:{Level:debug Format:logfmt} APICfg:{AllowedOrigin:^(?:.*)$ 
  ReadOnly:false AdminAPIEnabled:false TelemetryPath:/metrics Auth:zzz EnableFeatures: EnabledFeaturesList:[] 
  MaxQueryTimeout:2m0s SubQueryStepInterval:1m0s} LimitsCfg:{targetMemoryFlag:{kind:0 value:80} 
  TargetMemoryBytes:858993459} ConfigFile:config.yml TLSCertFile: TLSKeyFile: HaGroupLockID:0 PrometheusTimeout:-1ns 
  ElectionInterval:5s Migrate:false StopAfterMigrate:false UseVersionLease:true InstallExtensions:false 
  UpgradeExtensions:false UpgradePrereleaseExtensions:false HAEnabled:false}"
level=info ts=2021-04-09T17:08:09.043Z caller=client.go:92 msg="Skipping migration"
level=error ts=2021-04-09T17:08:09.083Z caller=client.go:122 msg="Unable to run migrations; failed to acquire the lock. 
  If the database is on an incorrect version, ensure there are no other connectors running and try again." 
  err="dependency error: Failed to acquire the migration lock to upgrade the schema version and unable to run with the
  old version. Please ensure that no other Promscale connectors with the old schema version are running. Received 
  schema version 0.0.0 but expected 0.3.0"
level=error ts=2021-04-09T17:08:09.084Z caller=runner.go:43 msg="aborting startup due to error" err="dependency error: 
  Failed to acquire the migration lock to upgrade the schema version and unable to run with the old version. Please 
  ensure that no other Promscale connectors with the old schema version are running. Received schema version 0.0.0 
  but expected 0.3.0"
```

There were no other Promscale instances running/connected at the time.

Are the required permissions documented somewhere, and if not, what are they?

**Comment by cevian** [comment]
Working on this now. Will provide more of an answer tomorrow.

**Comment by cevian** [comment]
work on a fix started in #587

**Comment by ssube** [comment]
Looks interesting, please let me know when you have something to test, I can give it a try.

**Comment by cevian** [comment]
@ssube Thanks. Should be ready in a few days.

**Comment by cevian** [comment]
#616 should be the last PR. Sorry took longer than expected. Plan to merge it probably tomorrow. The next release will definitely have it and is scheduled for May 17

**Comment by cevian** [comment]
Sorry I missed this. This has been available since release [0.4.0](https://github.com/timescale/promscale/releases/tag/0.4.0)
